File: jun/2025.05/250507_시리얼 번호.py
"""
[BOJ] 시리얼 번호 / 실버3
"""
n = int(input())

serial_numbers = []
for _ in range(n):
    serial_numbers.append(input())

serial_numbers.sort(key=lambda x: (len(x), sum(int(num) for num in x if num.isdigit()), x))

for number in serial_numbers:
    print(number)

File: yyong/2025.05/250507_시리얼_번호.js
const getSum = (str) => {
  return str
    .split("")
    .filter((char) => char >= "0" && char <= "9")
    .reduce((sum, char) => sum + Number(char), 0);
};

const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const N = Number(input[0]);

const serialNumber = [];

for (let i = 1; i < N + 1; i++) {
  serialNumber.push(input[i]);
}

serialNumber.sort((a, b) => {
  // 조건1
  if (a.length !== b.length) {
    return a.length - b.length;
  }

  // 조건2
  const sumA = getSum(a);
  const sumB = getSum(b);
  if (sumA !== sumB) {
    return sumA - sumB;
  }

  // 조건3
  return a.localeCompare(b);
});

console.log(serialNumber.join("\n"));
